Ai

Gr8-m8 opened the pull request.

Gr8-m8 committed:
Grunder

Gr8-m8 committed:
*Cell(basfunktion), *CellManager(basfunktion)

Gr8-m8 committed:
*Compability (Cell, Cellmanager)

17jelu committed:
Game* (DEBUG for Grid.cs), Grid* (To be reviewed)

Gr8-m8 committed:
*Cell, +GameObjekt

17jelu committed:
Game* (Grid Syntax), Grid* (Added TODO)

17jelu committed:
Game* (DEBUG Circle), Circle* (Opened Render)

17jelu commented on Gymnasiearbete/Cell.cs:
```diff
@@ -7,14 +7,54 @@
+        }
+    }
+
+    class food : GameObject
```
Ändra `food` till `Food`

17jelu committed:
Merge branch 'Graphics' of https://github.com/17jelu/Gymnasiearbete into Graphics

17jelu committed:
Merge branch 'AI' into Graphics

Gr8-m8 committed:
*food("Food")

17jelu committed:
Merge branch 'AI' of https://github.com/17jelu/Gymnasiearbete into AI

17jelu committed:
*Circle (UnitCircle), *Game (Circle DEBUG)

Gr8-m8 committed:
*Cellmanager(grid start), *Cell(perception start)

Gr8-m8 committed:
-Sector, *CellManager(sector Dictionary), Cell(PerceptionCheck)

Gr8-m8 committed:
Merge branch 'Graphics' into AI

Gr8-m8 committed:
*CellManager (Ger celler alla nödvändiga sectorer), *Cell (Simpel AI, detection range)

Första Iterationen av AI, är endast intresserad av det närmaste objektet.
I fallet att det är mat går den mot,
I fallet att de är cell, går den ifrån.
I fallet att det är inget går den Nord-Öst

Gr8-m8 committed:
*Grid(tilewidth), *Game(Cell Manager instans), *CellManager(sectorsize, detectionrange), *Cell(idledirection, *math)

Grid; tilewidth korrensponderar till sectorstorlek

Gr8-m8 committed:
*

Gr8-m8 committed:
*Cell(+EnergyManagement, *Collision), *Cellmanager(random food, *ofixerad updatering)

Gr8-m8 committed:
Game1(*debugCell), *Cellmanager(*Reproduce), *Cell(*Reproduce, *int to float)

Gr8-m8 committed:
Print Debug Sector

Gr8-m8 committed:
*Content, +Spritefont, *Cell(Collision, Reproduction, EnergyManagement), *CellManager(*Debugsector), Game1(+debugmessage)

Gr8-m8 committed:
Cell(Energy), CellManager(Energy Death drop), Game1(Experimental Cell 3)

Gr8-m8 committed:
*Game1(civ time, restart),  *Cellmanager(-cell food drop, food spawn timer), *Cell(mutaton, constant)

17jelu committed:
Merge branch 'AI' of https://github.com/17jelu/Gymnasiearbete into AI

Gr8-m8 committed:
+AI_Types, *Cell(-AI), *CellManager(sector point key), +Gameobject(grundklass för spelplansobjekt, +Entity)

Gr8-m8 committed:
protected var

17jelu committed:
Merge branch 'AI' of https://github.com/17jelu/Gymnasiearbete into AI

Gr8-m8 committed:
Debug(debugstring), AI(*mutation chance, *mutation overload)

Gr8-m8 committed:
*CellManager(Cell in sector by size)

Gr8-m8 committed:
AI(DEBUG, +Idledirection, +Actions), Cell(DEBUG, -Idledirection, -Actions), CellManager(simulationarea, DEBUG), Gameobject(DEBUG), Game1(DEBUG)

Gr8-m8 committed:
CellManager(debug), Game1(simulationarea), GameObject(+P)

Gr8-m8 committed:
Instansiate Simulationarea

Gr8-m8 committed:
+Memory()

Gr8-m8 committed:
include Memory()

Gr8-m8 committed:
AI(+idleDestination, *Desision, +MemoryReward), Cell(Memoryreward, Reproduce), CellManager(meoryreward)

Gr8-m8 committed:
+Sectorcontent(List Food, List Cell, List Other, Dictionary<Cell, CellDestination>), CellManager(*Sectorcontent), AI(return CellDestination)

Gr8-m8 committed:
CM.Initilize()

Gr8-m8 committed:
Sectorcontetn(Content.Content låter underligt -Content +All)

Gr8-m8 committed:
AI(DEBUGNOMEMORYSAVE)

Gr8-m8 committed:
Game1(sort Initilize)

Gr8-m8 committed:
Game1(auto restart), Grid(simulationarea)

17jelu committed:
Merge branch 'AI' into Graphics-2.1

17jelu committed:
*Sectorcontent(Renamed to SectorContent)

17jelu committed:
Actual Creation of Graphics 2.1

17jelu committed:
*CellManager(Made sectors accessable)

17jelu committed:
*SectorContent(Changed methods Foods/Cells/Others to Properties)

17jelu committed:
+StaticGlobals(Added Screen)

17jelu committed:
*CellManager(Fixed error caused by changing Cells method to a property)

17jelu committed:
*Camera(Removed X/Y & Added Zoom)

17jelu committed:
+Render(Supposed to be handling rendering)

17jelu committed:
*Game1(+SGScreen.Initialize & WIP Rendering)

17jelu committed:
*Game1.Update(Correctly centered spectating cell, renders food), *Camera.Initialize(set zoom to 1f)

17jelu committed:
*Game1.Update(Fixed center pos with zooming)

Gr8-m8 committed:
-Simulationarea, *Cell(Init)

Gr8-m8 committed:
-tock

Gr8-m8 committed:
AI(+Player "AI", *Debug)

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

17jelu committed:
*Render(Added Rendering), *Game(Removed Debug Rendering to Render.cs)

17jelu committed:
+DrawObject(Base class to Cricle, GraphicRectangle, Line etc...)

17jelu committed:
*CellManager(Commented out Draw method)

17jelu committed:
*Camera(Changed to struct & +GetRelativePos())

Changing Camera.cs to a struct and making the camera movement smooth.

17jelu committed:
*Line(Fixed line to 1 line)

17jelu committed:
*Render(Removed black dot and optimized Line)

17jelu committed:
*Render(Removed unneccessary color assignment)

Gr8-m8 committed:
*

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

Gr8-m8 committed:
staticGlobal(+Random), AI(idledestination maxDist), Game1(DEBUG)

17jelu committed:
*Render(Renders beta grid), *Line(Added option to disable relative camera for debug purpose)

17jelu committed:
Merge branch 'AI' into Graphics-2.1

17jelu committed:
*Camera(Capped Zoom)

17jelu committed:
*Render(Fixed grid)

17jelu committed:
Revert "*Render(Fixed grid)"

This reverts commit 349bc2430ee3e40f8903dc75b08c5ecb3299654a.

17jelu committed:
*Render(Fixed grid(1/2))

17jelu committed:
*Game(Added SGScreen Resize), *SGScreen(Added Resize Method)

17jelu committed:
*Render(Fixed grid(2/2))

Gr8-m8 committed:
AI(alllearning), Game1(*restart)

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

17jelu committed:
*Render(Fixed out of cam rendering)

Gr8-m8 committed:
AI(+*Data, +AI_PointsTargeting), Cell(Food lifetime), Cellmanaget(*Update(Cell/Food), remove empty sector), SectorContent(*All List, - cellDestination), DEBUG

17jelu committed:
-Grid.cs

17jelu committed:
*Cell(+Energy property)

17jelu committed:
*Camera(Added freecam)

17jelu committed:
-Grid.cs, *Game(Added pre alpha freecam)

17jelu committed:
*SGScreen(Fixed resize)

17jelu committed:
*SG(Changed SGScreen & SGBasicEffect to structs)

17jelu committed:
+CustomKeyboard

17jelu committed:
+CustomInputHandler, *CustomKeyboard(Made child of CustomInputHandler)

Gr8-m8 committed:
StaticGlobal(+float Clamp())

Gr8-m8 committed:
Cell(*Energy, *MemoryReward), GameObject(accessable Energy), CellManager(debug, *MemoryReward)

Gr8-m8 committed:
AI(*Kommentar, *Struktur)

Gr8-m8 committed:
StaticGlobal(AIControlls, CellManagerControlls), AI(spawndirection Normalize)

Gr8-m8 committed:
StaticGlobal(+EnergyControlls, +DefaultCellDNA), *CellSpeed * 10, AI(PointsTargeting last intresst)

Gr8-m8 committed:
*AI

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

17jelu committed:
Moved CustomInputHandler from CustomKeyboard.cs to CustomInputHandler.cs, *StaticGlobals(Added Keyboard & Mouse)

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

17jelu committed:
Revert "Moved CustomInputHandler from CustomKeyboard.cs to CustomInputHandler.cs, *StaticGlobals(Added Keyboard & Mouse)"

This reverts commit 15ec5ccb70235b4bd5a5988e29995b1f54587661.

17jelu committed:
Revert "Revert "Moved CustomInputHandler from CustomKeyboard.cs to CustomInputHandler.cs, *StaticGlobals(Added Keyboard & Mouse)""

This reverts commit 23b6e43914f12f3b5c30413912f2c9ffdc7b5a55.

17jelu committed:
*Game(DEBUG_PURPOSE, Added Mouse & Keyboard input)

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

Gr8-m8 committed:
Game1(*keys, Debug)

Gr8-m8 committed:
temp Css UIElement fix

17jelu committed:
+StaticGlobal.Family(Contains methods related to AI Family), *AI(Added new family)

17jelu committed:
*AI(Actually added new family) // Last commit didnt

Gr8-m8 committed:
Staticglobal(*EnergyControlls)

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

17jelu committed:
*Camera(+SpectatingCell, Simplified Initialize), *Game(Debug Camera)

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

17jelu committed:
*Cell(Added cell in Initialize, +ChangeSpectatingCell)

Gr8-m8 committed:
*

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

17jelu committed:
*StaticGlobal.Family(Fixed random index)

Gr8-m8 committed:
*

Gr8-m8 committed:
Merge branch 'Graphics-2.1' of https://github.com/17jelu/Gymnasiearbete into Graphics-2.1

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

Gr8-m8 committed:
*CM, StaticGlobal(Lore)

17jelu committed:
*StaticGlobals.Family(Changed families to Dictionary from List, Changed colors, +AddMember(family) & +KillMember(family))

Gr8-m8 committed:
Merge branch 'Graphics-2.1' of https://github.com/17jelu/Gymnasiearbete into Graphics-2.1

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

17jelu committed:
*Circle(Added Outline, Optimized code for vertex placement in instantiation), *GraphicRectangle(-BasicEffect in instantation), *StaticGlobals.Family(Names: +Black & +White), *Render(Renders Cells with their respective family name color)

17jelu committed:
*Render(Renders Cells with their respective family name color) | Last Render push was wrong

Gr8-m8 committed:
StaticGlobal(+Lore, +CM, +EnergyControlls, +AI controlls), Game1(*StaticGlobal.CM), Cellmanager(+Initilize, *Staticglobal.RAndom), Cell(Food lifetime), AI(StaticGlobal.Random)

Gr8-m8 committed:
Merge branch 'Graphics-2.1' of https://github.com/17jelu/Gymnasiearbete into Graphics-2.1

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

Gr8-m8 committed:
CellManager(*Pause, *bool SimulationEnd, *Initilize), StaticGlobal(CM), Game1(Key "{ }")

Gr8-m8 committed:
AI(family, Memory directory, Desicion[reproduce, stop]), Cell(staticglobal.random, *reproduce), CellManager(reproduce), Staticglobal(Clamp())

Gr8-m8 committed:
civilizationtime

17jelu committed:
Removed CSS once and for all

17jelu committed:
StaticGlobal(Added Save & Shutdown features), *StaticGlobal(Optmizied properties), -SGScreen, -SGEffect, StaticGlobal(+Scren, +BasicEffect), *Render(Removed comments), REST (Switched respective code to StaticGlobal)

17jelu committed:
*CustomKeyboard(+IsKeyReleased), +UIElement, *DrawObject(Added outline, optimized code)

17jelu committed:
*Game(BIG DEBUG)

17jelu committed:
Merge branch 'Graphics-2.1' of https://github.com/17jelu/Gymnasiearbete into Graphics-2.1

Gr8-m8 committed:
Merge branch 'Graphics-2.1' into AI

Gr8-m8 committed:
+UIElement, Renderer(*SectorSize)
